message.go

```diff
@@ -915,7 +1016,15 @@ func (m *Message) unmarshalField(field reflect.Value, rv reflect.Value) error {
 			return fmt.Errorf("%v: %v", errUnmarshalNonMessage, rv.Type())
 		}
 
-		return msg.unmarshal(field.Interface())
+		fp := field
+		if field.IsNil() {
+			fp = reflect.New(field.Type().Elem())
+			field.Set(fp)
+		}
+
+		if err := msg.unmarshal(fp.Interface()); err != nil {
```

Comment:
What's the need for the `if` statement here? Why not just `return msg.unmarshal(fp.interface())`?

Reply:
Not longer needed. I believe I re-ordered something at one point here.